app/services/store_participant_eligibility.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+class StoreParticipantEligibility < BaseService
+  attr_reader :participant_profile, :eligibility_options
+
+  def initialize(participant_profile:, eligibility_options: {})
+    @participant_profile = participant_profile
+    @eligibility_options = eligibility_options
+  end
+
+  def call
+    participant_eligibility = ECFParticipantEligibility.find_or_initialize_by(participant_profile: participant_profile)
+
+    previous_status = participant_eligibility.status
+    previous_reason = participant_eligibility.reason
+
+    update_and_store_eligibility_values!(participant_eligibility)
+
+    # send rejection notifications
+    if ((participant_eligibility.ineligible_status? && previous_status != "ineligible") ||
```

Comment:
Could this be made more readable? Perhaps `if became_ineligible? || participant_eligibility.ineligible_status? && reason_changed?`